components/provisioner/internal/util/retry.go

```diff
@@ -0,0 +1,22 @@
+package util
+
+import (
+	"time"
+
+	"github.com/kyma-project/control-plane/components/provisioner/internal/apperrors"
+
+	"github.com/sirupsen/logrus"
+)
+
+func RetryOnError(interval time.Duration, count int, errMsgFmt string, function func() apperrors.AppError) apperrors.AppError {
```

Comment:
There is a numer of libraries that implement retries (e.g. github.com/avast/retry-go,  k8s.io/apimachinery/pkg/util/wait).What is the reason we implemented this function from scratch?

Reply:
We want to return apperrors.AppError from Director calls